src/test/scala/invertible/composed.scala

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright 2014 - 2016 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package invertible
+
+import invertible._
+import matryoshka._, TraverseT._, Recursive.ops._, Corecursive.ops._
+import scalaz._, Scalaz._
+
+/*
+Syntax for a simple expression language, composed out of several distinct
+syntaxes via Coproducts.
+
+Here, matryoshka's Fix is used to turn the functors into recursive types (compare to
+`example.scala`, where Cofree is used in a similar way, but also captures the source
+position as an annotation on each node.)
+*/
+
+object Composed {
+
+  // These constructors are each independent:
+  final case class Num[A](value: Int)
+  final case class Add[A](left: A, right: A)
+  final case class Mul[A](left: A, right: A)
+
+  // These two constructors go together, so extend a common trait:
+  sealed trait Bound[A]
+  final case class Let[A](sym: Symbol, expr: A, body: A) extends Bound[A]
+  final case class Free[A](sym: Symbol) extends Bound[A]
+
+
+  //
+  // Iso instances for each constructor:
+  //
+
+  import Iso._
+
+  def num[A] = total[Int, Num[A]](Num(_), _.value)
+  def add[A] = iso[(A, A), Add[A]](
+    { case (l, r) => Add(l, r) },
+    { case Add(l, r) => (l, r) })
+  def mul[A] = iso[(A, A), Mul[A]](
+    { case (l, r) => Mul(l, r) },
+    { case Mul(l, r) => (l, r) })
+
+  def let[A] = iso[(Symbol, (A, A)), Bound[A]](
+    { case (sym, (expr, body)) => Let(sym, expr, body) },
+    { case Let(sym, expr, body) => (sym, (expr, body)) })
+  def free[A] = iso[Symbol, Bound[A]](
+    { case sym => Free(sym) },
+    { case Free(sym) => sym })
+
+
+  //
+  // Some utilities:
+  //
+
+  def fixF[F[_], G[_]](implicit I: F :<: G): Iso[F[Fix[G]], Fix[G]] =
```

Comment:
Ok, not that this’ll help inference issues, but I think a few things could be done here …

Rename `Iso` to `PIso` (or `IsoM[M[_]]`) and then have simple transformations `monocle.Iso => PIso` and `monocle.Prism => PIso`. Then there could be a prism like `coproduct(implicit I: F :<: G): Prism[G[A], F[A]]`, and then `coproduct.reverse composeIso matryoshka.recCorecIso` … or something … would give you a Prism that you could convert to a PIso and call `fixF`.